rasgoql/rasgoql/data/postgres.py

```diff
@@ -0,0 +1,630 @@
+        `overwrite`: bool
+            pass True when this table name already exists in your DataWarehouse
+            and you know you want to overwrite it
+            WARNING: This will completely overwrite data in the existing table
+        """
+        table_type = check_table_type(table_type)
+        fqtn = magic_fqtn_handler(fqtn, self.default_namespace)
+        pg_namespace = self._pg_appropriate_namespace(fqtn=fqtn)
+        if self._table_exists(fqtn=fqtn) and not overwrite:
+            msg = f'A table or view named {fqtn} already exists. ' \
+                   'If you are sure you want to overwrite it, ' \
+                   'pass in overwrite=True and run this function again'
+            raise TableConflictException(msg)
+        query = f"CREATE OR REPLACE {table_type} {pg_namespace} AS {sql}"
+        self.execute_query(query, acknowledge_risk=True, response='None')
+        return fqtn
+
+    @property
+    def default_namespace(self) -> str:
+        """
+        Returns the default database.schema of this connection
+        """
+        return f'{self.default_database}.{self.default_schema}'
+
+    def execute_query(
+            self,
+            sql: str,
+            response: str = 'tuple',
+            acknowledge_risk: bool = False
+        ):
+        """
+        Run a query against Postgres and return all results
+
+        `sql`: str:
+            query text to execute
+        `response`: str:
+            Possible values: [dict, df, None]
+        `acknowledge_risk`: bool:
+            pass True when you know your SQL statement contains
+            a potentially dangerous or data-altering operation
+            and still want to run it against your DataWarehouse
+        """
+        response = check_response_type(response)
+        if is_scary_sql(sql) and not acknowledge_risk:
+            msg = 'It looks like your SQL statement contains a ' \
+                  'potentially dangerous or data-altering operation.' \
+                  'If you are positive you want to run this, ' \
+                  'pass in acknowledge_risk=True and run this function again.'
+            raise SQLWarning(msg)
+        verbose_message(
+            f"Executing query: {sql}",
+            logger
+        )
+        if response == 'DICT':
+            return self._query_into_dict(sql)
+        if response == 'DF':
+            return self._query_into_df(sql)
+        return self._execute_string(sql, ignore_results=(response == 'NONE'))
+
+    # TODO: Note: DDL isn't returnable by objects while querying postgres. 
+    def get_ddl(
+            self,
+            fqtn: str
+        ) -> pd.DataFrame:
+        """
+        Returns a DataFrame describing the column in the table
+
+        `fqtn`: str:
+            Fully-qualified Table Name (database.schema.table)
+        """
+        fqtn = magic_fqtn_handler(fqtn, self.default_namespace)
+        _, schema_name, table_name = parse_fqtn(fqtn)
+        sql = (
```

Comment:
This probably answers our question re whether this class can be an ANSI class or Postgres-specific?